tests/trace/dsl/test_query.py

```diff
@@ -1,302 +1,768 @@
+            "Latency (milliseconds)": [30000.0],
+        }
+    ).set_index("context.span_id")
+    actual = sq(session, root_spans_only=True)
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1),
+        expected.sort_index().sort_index(axis=1),
+    )
+    del sq, actual, expected
 
 
-def test_explosion(spans):
-    explode = Explosion("12")
-    assert list(explode(spans[2])) == [
-        {
-            "12": 1,
-            "context.span_id": "2",
-            "position": 0,
-        },
+def test_root_spans_only(session: Session) -> None:
+    sq = SpanQuery().select(
+        "name",
+        **{"Latency (milliseconds)": "latency_ms"},
+    )
+    expected = pd.DataFrame(
         {
-            "12": 2,
-            "context.span_id": "2",
-            "position": 1,
-        },
-    ]
+            "context.span_id": ["234"],
+            "name": ["root span"],
+            "Latency (milliseconds)": [30000.0],
+        }
+    ).set_index("context.span_id")
+    actual = sq(
+        session,
+        project_name="abc",
+        root_spans_only=True,
+    )
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1),
+        expected.sort_index().sort_index(axis=1),
+    )
+    del sq, actual, expected
 
-    explode = Explosion(RETRIEVAL_DOCUMENTS)
 
-    assert list(explode(spans[0])) == []
-    assert list(explode(spans[1])) == [
+def test_start_time(session: Session) -> None:
+    sq = SpanQuery().select("name")
+    expected = pd.DataFrame(
         {
-            DOCUMENT_CONTENT: "10",
-            DOCUMENT_SCORE: 100,
-            "context.span_id": "1",
-            "document_position": 0,
+            "context.span_id": ["567"],
+            "name": ["llm span"],
         }
-    ]
-    assert list(explode(spans[2])) == [
+    ).set_index("context.span_id")
+    actual = sq(
+        session,
+        project_name="abc",
+        start_time=datetime.fromisoformat(
+            "2021-01-01T00:00:20.000+00:00",
+        ),
+    )
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1),
+        expected.sort_index().sort_index(axis=1),
+    )
+    del sq, actual, expected
+
+
+def test_stop_time(session: Session) -> None:
+    sq = SpanQuery().select("name")
+    expected = pd.DataFrame(
         {
-            DOCUMENT_CONTENT: "20",
-            "context.span_id": "2",
-            "document_position": 0,
-        },
+            "context.span_id": ["234", "345"],
+            "name": ["root span", "embedding span"],
+        }
+    ).set_index("context.span_id")
+    actual = sq(
+        session,
+        project_name="abc",
+        stop_time=datetime.fromisoformat(
+            "2021-01-01T00:00:01.000+00:00",
+        ),
+    )
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1),
+        expected.sort_index().sort_index(axis=1),
+    )
+    del sq, actual, expected
+
+
+def test_filter_for_none(session: Session) -> None:
```

Comment:
Same comment here about splitting into multiple test cases.

Reply:
Looks like there are many tests on filtering in this file. Do these belong in `test_filter.py`?

Reply:
Yea they do, but they belong here too. So I opted to keep them here for more "holistic" testing and leave the more syntactical stuff over there.